rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java

```diff
@@ -119,12 +160,26 @@ private void setValidProductStatus() {
     }
 
     private void setResultAndFinish(OneScanHelper.ScanResponse response) {
+
         Intent resultIntent = new Intent();
-        Barcode barcode = new Barcode();
-        barcode.displayValue = StringUtils.join(new String[]{response.serialNumber, response.expirationDate,
-                response.lot, response.productId, String.valueOf(response.sensorTriggered),
-                response.status}, ',');
-        resultIntent.putExtra(BARCODE_KEY, barcode);
+
+        if (enableBatchScan) {
+            try {
+                JSONObject dataObject = new JSONObject();
+                dataObject.put("scans", dataArray);
+                resultIntent.putExtra("data", dataObject.toString());
+            } catch (JSONException e) {
+                e.printStackTrace();
```

Comment:
Could this be replaced with `Timber.e(e)`?

Reply:
updated.